zkir/Dialect/Field/IR/FieldTypes.cpp

```diff
@@ -48,6 +47,12 @@ unsigned getIntOrPrimeFieldBitWidth(Type type) {
   return cast<IntegerType>(type).getWidth();
 }
 
+mod_arith::ModArithType convertPrimeFieldType(PrimeFieldType type) {
+  IntegerAttr modulus = type.getModulus();
+  bool isMontgomery = type.isMontgomery();
+  return mod_arith::ModArithType::get(type.getContext(), modulus, isMontgomery);
+}
```

Comment:
I think this is specific to `FieldToModArith` pass? In this sense, the `FieldOperation` should maybe move to `FieldToModArith` as well.